### PR TITLE
Redux 1

### PR DESCRIPTION
**Pull request by SaraIsabelSantacruz**
## Summary
- all books lsited
- search implemented
- button for add books implemented

## Screenshot
redux-1

## Trello Card
https://trello.com/c/LUsCUB7f

### CODE REVIEW

File: react/finger-exercises/redux/02 - react-redux/src/app/screens/App/index.js [hunk 1]
```diff
@@ -15,24 +18,54 @@ class App extends Component {
...
+    );
+  };
 
   // TODO to implement the dispatch
-  addItem = itemId => {};
+  addItem = itemId => {
+    const { bookSelected } = this.state;
+    const newBook = bookSelected.map(book => {
+      if (book.id === itemId) {
+        const cont = book.quantity + 1;
```
Comment (gon-escalante) [inline react/finger-exercises/redux/02 - react-redux/src/app/screens/App/index.js]: you can do 
```
const quantity = book.quantity + 1;
book = { ...book, quantity };
```

File: react/finger-exercises/redux/02 - react-redux/src/app/screens/App/index.js [hunk 2]
```diff
@@ -15,24 +18,54 @@ class App extends Component {
...
+      }
+      return book;
+    });
+    store.dispatch(actionsCreators.addToCart(newBook));
+  };
 
   // TODO to implement the dispatch
-  removeItem = itemId => {};
+  removeItem = itemId => {
+    const { bookSelected } = this.state;
```
Comment (gon-escalante) [inline react/finger-exercises/redux/02 - react-redux/src/app/screens/App/index.js]: i'd also move this logic to the reducer

File: react/finger-exercises/redux/02 - react-redux/src/app/screens/App/index.js [hunk 3]
```diff
@@ -1,7 +1,10 @@
 import React, { Component, Fragment } from 'react';
+import PropTypes from 'prop-types';
 import store from '@redux/store';
+import actionsCreators from '@redux/book/actions';
```
Comment (CrisVega08) [inline react/finger-exercises/redux/02 - react-redux/src/app/screens/App/index.js]: On some component you can use various actions so I recommend change this name for something more descriptive for example bookActions

File: react/finger-exercises/redux/02 - react-redux/src/app/screens/App/index.js [hunk 4]
```diff
@@ -15,24 +18,44 @@ class App extends Component {
...
+    const { data } = this.props;
     store.subscribe(() => {
-      const { books, bookSelected } = store.getState();
-      this.setState({ books, bookSelected });
+      const { books, originalData, bookSelected } = store.getState();
+      this.setState({ bookSelected });
+      if (books.length > 0) this.setState({ books });
+      else this.setState({ books: originalData });
     });
     // TODO to implement the dispatch
```
Comment (CrisVega08) [inline react/finger-exercises/redux/02 - react-redux/src/app/screens/App/index.js]: Delete this line

File: react/finger-exercises/redux/02 - react-redux/src/app/screens/App/index.js [hunk 5]
```diff
@@ -15,24 +18,44 @@ class App extends Component {
...
+    store.dispatch(actionsCreators.getBooks(data));
   }
 
   // TODO to implement the dispatch
```
Comment (CrisVega08) [inline react/finger-exercises/redux/02 - react-redux/src/app/screens/App/index.js]: Same here

File: react/finger-exercises/redux/02 - react-redux/src/app/screens/App/index.js [hunk 6]
```diff
@@ -1,7 +1,10 @@
 import React, { Component, Fragment } from 'react';
+import PropTypes from 'prop-types';
```
Comment (CrisVega08) [inline react/finger-exercises/redux/02 - react-redux/src/app/screens/App/index.js]: I suggest importing the property types that you use in the component, for example, in this file only use arrayOf 
```
import { arrayOf } from 'prop-types';
```
This makes the code more readable.

File: react/finger-exercises/redux/02 - react-redux/src/app/screens/App/index.js [hunk 7]
```diff
@@ -15,24 +18,44 @@ class App extends Component {
...
 
   // TODO to implement the dispatch
-  onSearch = value => {};
+  onSearch = value => {
+    const { books } = this.state;
+    const result = value && books.filter(book => book.name.toLowerCase().includes(value.toLowerCase()));
+    store.dispatch(actionsCreators.searchBook(result));
+  };
 
   // TODO to implement the dispatch
```
Comment (CrisVega08) [inline react/finger-exercises/redux/02 - react-redux/src/app/screens/App/index.js]: Same here

File: react/finger-exercises/redux/02 - react-redux/src/app/screens/App/index.js [hunk 8]
```diff
@@ -75,4 +98,8 @@ class App extends Component {
   }
 }
 
+App.propTypes = {
+  data: PropTypes.arrayOf(listBooksPropType)
```
Comment (CrisVega08) [inline react/finger-exercises/redux/02 - react-redux/src/app/screens/App/index.js]: If you import directly arrayOf propType here you can change this for something like.

```
 data: arrayOf(listBooksPropType)
```

File: react/finger-exercises/redux/02 - react-redux/src/index.js [hunk 9]
```diff
@@ -2,8 +2,9 @@ import React from 'react';
 import ReactDOM from 'react-dom';
 import App from '@screens/App';
 
+import { DATA } from './constants/data';
```
Comment (CrisVega08) [inline react/finger-exercises/redux/02 - react-redux/src/index.js]: I recommend move this import to screens/app/index or storybook/actions

File: react/finger-exercises/redux/02 - react-redux/src/redux/book/actions.js [hunk 10]
```diff
@@ -1,5 +1,3 @@
-import { DATA } from '@constants/data';
```
Comment (CrisVega08) [inline react/finger-exercises/redux/02 - react-redux/src/redux/book/actions.js]: Why do you deleted it ?

File: react/finger-exercises/redux/02 - react-redux/src/constants/propTypes.js [hunk 11]
```diff
@@ -5,3 +5,12 @@ export const bookSelectedPropType = shape({
   summary: string,
   quantity: number
 });
+
+export const listBooksPropType = shape({
```
Comment (gon-escalante) [inline react/finger-exercises/redux/02 - react-redux/src/constants/propTypes.js]: you are not using this proptype

**Pull request merged**